File: Benchmarking/src/gen164.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  struct s1 {
    std::vector<s0> p0;
    s0 p1;
    s1(std::vector<s0> v0, s0 v1): p0(v0), p1(v1) { }
  };
  struct s4 {
    double p0;
    s1 p1;
    s4(double v0, s1 v1): p0(v0), p1(v1) { }
  };
  s0 f55(const s0 &v0) {
    s0 v4 = v0;
    const std::vector<std::vector<double>> v6 = v4.p0;
    s0 v1 = v4;
    v1.p0 = v6;
    const std::vector<std::vector<double>> v10 = v1.p0;
    v1 = v4;
    const std::vector<std::vector<double>> v5 = v4.p1;
    const std::vector<std::vector<double>> v15 = v1.p0;
    v1.p1 = v5;
    const std::vector<s0> v24 { v0, v0, v1, v1, v1, v1, v1 };
    const s0 v23 = v24[6];
    const std::vector<std::vector<s0>> v25 { v24 };
    const std::vector<std::vector<double>> v22 = v0.p0;
    v1.p0 = v22;
    v4.p1 = v6;
    const std::vector<s0> v47 = v25[0];
    v4.p0 = v6;
    const std::vector<std::vector<double>> v34 = v1.p1;
    const std::vector<std::vector<std::vector<s0>>> v68 { v25, v25, v25 };
    std::vector<s0> v45 = v47;
    std::vector<s0> v48 = v45;
    const std::vector<std::vector<s0>> v33 = v68[0];
    std::vector<std::vector<s0>> v43 = v33;
    v4.p1 = v10;
    v1.p0 = v15;
    s0 v82 = v23;
    v45[4] = v82;
    const std::vector<std::vector<double>> v60 = v82.p0;
    v43[0] = v48;
    std::vector<std::vector<s0>> v93 = v43;
    v82.p0 = v60;
    const std::vector<s0> v66 = v93[0];
    v1.p0 = v34;
    const s0 v89 = v66[1];
    v4.p0 = v34;
    return v89;
  }
  s0 f17(const s0 &v0) {
    const s0 v8 = f55(v0);
    const s0 v12 = f55(v8);
    const std::vector<s0> v13 { v12, v12, v0, v0, v12, v8, v12 };
    const s0 v22 = v13[1];
    const s0 v18 = v13[1];
    const s0 v50 = v13[4];
    const std::vector<s0> v21 { v12, v0, v50, v22, v18 };
    const s0 v83 = v21[1];
    return v83;
  }
  __attribute__((noinline))
  double f0(const s0 &v0, const s0 &v1, const s4 &v2, const s4 &v3, const double &v4) {
    const s0 v5 = f17(v0);
    const double v8 = v2.p0;
    const double v9 = v4 - v8;
    const std::vector<std::vector<double>> v16 = v5.p1;
    const std::vector<double> v14 = v16[2];
    std::vector<double> v69 = v14;
    const double v52 = v69[0];
    const double v137 = v52 + v9;
    return v137;
  }
  int main() {
    s0 v0({ { { 0.0 }, { 1.0 }, { 2.0 } }, { { 3.0 }, { 4.0 }, { 5.0 } } });
    s0 v1({ { { 6.0 }, { 7.0 }, { 8.0 } }, { { 9.0 }, { 10.0 }, { 11.0 } } });
    s4 v2({ 12.0, { { { { { 13.0 }, { 14.0 }, { 15.0 } }, { { 16.0 }, { 17.0 }, { 18.0 } } }, { { { 19.0 }, { 20.0 }, { 21.0 } }, { { 22.0 }, { 23.0 }, { 24.0 } } }, { { { 25.0 }, { 26.0 }, { 27.0 } }, { { 28.0 }, { 29.0 }, { 30.0 } } } }, { { { 31.0 }, { 32.0 }, { 33.0 } }, { { 34.0 }, { 35.0 }, { 36.0 } } } } });
    s4 v3({ 37.0, { { { { { 38.0 }, { 39.0 }, { 40.0 } }, { { 41.0 }, { 42.0 }, { 43.0 } } }, { { { 44.0 }, { 45.0 }, { 46.0 } }, { { 47.0 }, { 48.0 }, { 49.0 } } }, { { { 50.0 }, { 51.0 }, { 52.0 } }, { { 53.0 }, { 54.0 }, { 55.0 } } } }, { { { 56.0 }, { 57.0 }, { 58.0 } }, { { 59.0 }, { 60.0 }, { 61.0 } } } } });
    double v4(62.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2, v3, v4);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen164.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: [s0]
    var p1: s0
  }
  struct s4 {
    var p0: Double
    var p1: s1
  }
  func f55(_ v0: s0) -> s0 {
    var v4: s0 = v0
    let v6: [[Double]] = v4.p0
    var v1: s0 = v4
    v1.p0 = v6
    let v10: [[Double]] = v1.p0
    v1 = v4
    let v5: [[Double]] = v4.p1
    let v15: [[Double]] = v1.p0
    v1.p1 = v5
    let v24: [s0] = [v0, v0, v1, v1, v1, v1, v1]
    let v23: s0 = v24[6]
    let v25: [[s0]] = [v24]
    let v22: [[Double]] = v0.p0
    v1.p0 = v22
    v4.p1 = v6
    let v47: [s0] = v25[0]
    v4.p0 = v6
    let v34: [[Double]] = v1.p1
    let v68: [[[s0]]] = [v25, v25, v25]
    var v45: [s0] = v47
    var v48: [s0] = v45
    let v33: [[s0]] = v68[0]
    var v43: [[s0]] = v33
    v4.p1 = v10
    v1.p0 = v15
    var v82: s0 = v23
    v45[4] = v82
    let v60: [[Double]] = v82.p0
    v43[0] = v48
    var v93: [[s0]] = v43
    v82.p0 = v60
    let v66: [s0] = v93[0]
    v1.p0 = v34
    let v89: s0 = v66[1]
    v4.p0 = v34
    return v89
  }
  func f17(_ v0: s0) -> s0 {
    let v8: s0 = f55(v0)
    let v12: s0 = f55(v8)
    let v13: [s0] = [v12, v12, v0, v0, v12, v8, v12]
    let v22: s0 = v13[1]
    let v18: s0 = v13[1]
    let v50: s0 = v13[4]
    let v21: [s0] = [v12, v0, v50, v22, v18]
    let v83: s0 = v21[1]
    return v83
  }
  @inline(never)
  func f0(_ v0: s0, _ v1: s0, _ v2: s4, _ v3: s4, _ v4: Double) -> Double {
    let v5: s0 = f17(v0)
    let v8: Double = v2.p0
    let v9: Double = v4 - v8
    let v16: [[Double]] = v5.p1
    let v14: [Double] = v16[2]
    var v69: [Double] = v14
    let v52: Double = v69[0]
    let v137: Double = v52 + v9
    return v137
  }
  func benchmark() {
    let v0: s0 = s0(p0: [[0.0], [1.0], [2.0]], p1: [[3.0], [4.0], [5.0]])
    let v1: s0 = s0(p0: [[6.0], [7.0], [8.0]], p1: [[9.0], [10.0], [11.0]])
    let v2: s4 = s4(p0: 12.0, p1: s1(p0: [s0(p0: [[13.0], [14.0], [15.0]], p1: [[16.0], [17.0], [18.0]]), s0(p0: [[19.0], [20.0], [21.0]], p1: [[22.0], [23.0], [24.0]]), s0(p0: [[25.0], [26.0], [27.0]], p1: [[28.0], [29.0], [30.0]])], p1: s0(p0: [[31.0], [32.0], [33.0]], p1: [[34.0], [35.0], [36.0]])))
    let v3: s4 = s4(p0: 37.0, p1: s1(p0: [s0(p0: [[38.0], [39.0], [40.0]], p1: [[41.0], [42.0], [43.0]]), s0(p0: [[44.0], [45.0], [46.0]], p1: [[47.0], [48.0], [49.0]]), s0(p0: [[50.0], [51.0], [52.0]], p1: [[53.0], [54.0], [55.0]])], p1: s0(p0: [[56.0], [57.0], [58.0]], p1: [[59.0], [60.0], [61.0]])))
    let v4: Double = 62.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 62.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2, v3, v4)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
